internal/service/users/invite.go

```diff
@@ -0,0 +1,11 @@
+package users
+
+import (
+	"context"
+
+	"github.com/ucl-arc-tre/portal/internal/types"
+)
+
+func (s *Service) InviteUser(ctx context.Context, email string, sponsor types.Sponsor) error {
```

Comment:
this should, in my opinion, be idempotent. so if a user has already been invited then they're not invited again (so persist in the database and check if they exist – if they don't then send invite and persist if successful)